test/python/transpiler/test_token_swapper.py

```diff
@@ -102,11 +103,15 @@ def test_large_partial_random(self) -> None:
         """Test a random (partial) mapping on a large randomly generated graph"""
         size = 100
         # Note that graph may have "gaps" in the node counts, i.e. the numbering is noncontiguous.
-        graph = nx.dense_gnm_random_graph(size, size ** 2 // 10)
-        graph.remove_edges_from((i, i) for i in graph.nodes)  # Remove self-loops.
+        graph = rx.undirected_gnm_random_graph(size, size ** 2 // 10)
+        for i in graph.node_indexes():
+            try:
+                graph.remove_edge(i, i)  # Remove self-loops.
+            except rx.NoEdgeBetweenNodes:
+                continue
         # Make sure the graph is connected by adding C_n
-        nodes = list(graph.nodes)
-        graph.add_edges_from((node, nodes[(i + 1) % len(nodes)]) for i, node in enumerate(nodes))
+        graph.add_edges_from_no_data(
+            [(i, i + 1) for i in range(len(graph) - 1)])
```

Comment:
What's the difference to `extend_from_edge_list`?

Reply:
`extend_from_edge_list()` will add nodes to the graph if there is a node index in the edge tuple that doesn't exist. While `add_edges_from_no_data()` will error if there is an index in the edge tuple that doesn't exist.

https://retworkx.readthedocs.io/en/stable/stubs/retworkx.PyGraph.extend_from_edge_list.html#retworkx.PyGraph.extend_from_edge_list